repo2docker/app.py

```diff
@@ -474,6 +475,18 @@ def _get_free_port(self):
         return port
 
     def start(self):
+        # Check if r2d can connect to docker daemon
+        if self.build:
+            try:
+                client = docker.APIClient(version='auto',
+                                          **kwargs_from_env())
+                del client
+            except DockerException as e:
+                print("Docker client initialization error. Check if docker is running in the host.")
```

Comment:
I think we should also print the actual error message from docker here, rather than swallow it. Also, version='auto' makes one HTTP request to the docker daemon - can we instead have this only show up where we actually connect?

Reply:
For printing the exact message ... will make the change. 
We can do that but that is not an early check to see if we have the ability to move forward with connecting to docker and building the docker image. We would have still ended up cloning the repo and then found out that we are not able to connect to docker. 

This way even though we are making additional request to docker daemon, we will be making sure that we do not clone the repo unless we know that we can build.

Reply:
Ah, fair enough. And checking for build means we can still do a '--no-build' without docker, which sounds good. I'll merge after the message is changed, thanks!

Reply:
done